### PR TITLE
Bug time input validation

### LINKED ISSUE #157: Time config input validation
Only allow numbers, no other characters

### PR DESCRIPTION
**Pull request by raymondhonsumchiu**
**Describe the issue**
Users could enter any character into the timer's input field when we want there to be only integer values.

**Describe the solution**
- Added a helper function in the TimerConfig component to validate the input against the following regular expression: ```/^\d+$/```, this ensures only numbers ranging from 0-9 are allowed. 
- When invalid numbers are inputted, the corresponding minutes/seconds state is set to NaN for another check in CurrentTask.jsx to prevent the start button from activating the timer when an invalid value is present.
- Finally, in TimerConfig.jsx, or the IntervalInput component more specifically, red warning labels were added to appear when a value is incorrect. 

image

**Possible Risks**
- When I ran ```npm install``` for some reason the package-lock.json removed a bunch of lines... though the application seems to work completely fine

### CODE REVIEW

File: src/components/current-task/TimerConfig.jsx [hunk 1]
```diff
@@ -59,30 +69,52 @@ function TimerConfig(props) {
   const { setTimerConfigValues } = props;
 
   const [isChecked, setIsChecked] = useContext(FullscreenContext);
+  const [isInvalidWorkTime, setIsInvalidWorkTime] = useState(false);
```
Comment (kcho9906) [inline src/components/current-task/TimerConfig.jsx:74]: Hooks should be declared at the start of the component

**Review by kcho9906 — CHANGES_REQUESTED** [review]
If you didn't need to add any modules to the project maybe we can restore this file and keep it as is - I seemed to have problems with the package lock as well. Perhaps we can open an issue to address it, but updating the package lock shouldn't be in this pull request in my opinion

File: src/components/current-task/TimerConfig.jsx (hunk 1 above)
Comment (raymondhonsumchiu) [reply src/components/current-task/TimerConfig.jsx:74]: This case was kind of weird, in TimerConfig.jsx they seem to define two components Timerconfig (where the hooks are located at the top of in this case) and a separate IntervalInput component above it, hence I just followed the "convention" that was already there.

**Comment by raymondhonsumchiu** [comment]
> If you didn't need to add any modules to the project maybe we can restore this file and keep it as is - I seemed to have problems with the package lock as well. Perhaps we can open an issue to address it, but updating the package lock shouldn't be in this pull request in my opinion

Oki doki, restored the old package-lock.json

File: src/components/current-task/TimerConfig.jsx (hunk 1 above)
Comment (kcho9906) [reply src/components/current-task/TimerConfig.jsx:74]: Ah, my mistake. I must've missed that

**Review by kcho9906 — APPROVED** [review]

**Comment by kcho9906** [comment]
> > If you didn't need to add any modules to the project maybe we can restore this file and keep it as is - I seemed to have problems with the package lock as well. Perhaps we can open an issue to address it, but updating the package lock shouldn't be in this pull request in my opinion
> 
> Oki doki, restored the old package-lock.json

Nice, everything looks fine to me - great work

**Review by vkoil — APPROVED** [review]
Looks good to go, just remember to accept both changes when resolving the conflicts with this merge.

**Pull request merged**